src/adc_dma.cpp

```diff
@@ -27,6 +27,15 @@
 #include "log.h"
 #include "pwm_switch.h"
 
+extern Charger charger;
```

Comment:
Why not put these extern statements in appropriate header file ( e.g. main.h )? These global variables are used in a couple of places.

Reply:
Yes, good idea.